hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/compact/ScheduleCompactionActionExecutor.java

```diff
@@ -213,4 +216,8 @@ private Long parsedToSeconds(String time) {
     return TimelineUtils.parseDateFromInstantTimeSafely(time).orElseThrow(() -> new HoodieCompactionException("Failed to parse timestamp " + time))
             .getTime() / 1000;
   }
+
+  private  BaseHoodieCompactionPlanGenerator createCompactionPlanGenerator(String planGeneratorClass, HoodieTable table, HoodieEngineContext context, HoodieWriteConfig config) {
```

Comment:
We already have the stratefy classes why need to customize the plan generator?

Reply:
CompactionStrategy takes in the list of compaction operations and decides which ones to pick it up based on strategy logic. This is for the ability to keep track of metadata from previous compaction results in extraMetadata and then decide the plan.